disco_aws_automation/disco_acm.py

```diff
@@ -69,15 +72,27 @@ def get_certificate_arn(self, dns_name):
             return None
 
         try:
-            cert_summary = self.acm.list_certificates()[CERT_SUMMARY_LIST_KEY]
-            certs = [cert for cert in cert_summary if self._in_domain(cert[DOMAIN_NAME_KEY], dns_name)]
-            # determine the most specific domain match
-            certs.sort(key=lambda cert: len(cert[DOMAIN_NAME_KEY]), reverse=True)
-            if not certs:
+            cert_summary = throttled_call(self.acm.list_certificates)[CERT_SUMMARY_LIST_KEY]
+
+            cert_matches = []
+            for cert in cert_summary:
+                response = throttled_call(self.acm.describe_certificate, CertificateArn=cert[CERT_ARN_KEY])
+                cert = response[CERT_KEY]
+                # search for the dns_name in the cert's alternative domain names (includes the main one)
+                for alt_name in cert[CERT_ALT_NAMES_KEY]:
+                    if self._in_domain(alt_name, dns_name):
+                        # a tuple of the matching cert domain name and the actual cert
+                        cert_matches.append((alt_name, cert))
```

Comment:
It looks like the main domain name is always in the alternatives list so there should always be one